sale_force_invoiced/wizards/sale_make_invoice_advance.py

```diff
@@ -0,0 +1,26 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class SaleAdvancePaymentInv(models.TransientModel):
+    _inherit = "sale.advance.payment.inv"
+
+    def _check_sale_orders_fully_invoiced(
+        self,
+    ):
+        for sale_order in self.sale_order_ids:
+            if sale_order.invoice_status == "invoiced":
```

Comment:
Shouldn't you test for `force invoice` instead ?
What if the SO is fully invoiced (without force flag) and you run the wizard?

Reply:
Updated code to test `force_invoiced`